tests/test_checker.py

```diff
@@ -52,6 +54,67 @@
 NUM_NEW_VERSIONS = 2
 
 
+class _TestWithInlineManifest(unittest.IsolatedAsyncioTestCase):
+    async def _test_update(
+        self,
+        filename,
+        contents,
+        expected_new_contents,
+        expected_updates,
+        expected_data_count=1,
+        new_release=True,
+        require_important_update=False,
+        test_important_update=False,
+    ):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            manifest = os.path.join(tmpdir, filename)
+            with open(manifest, "w") as f:
+                f.write(contents)
+
+            appdata = os.path.join(
+                tmpdir,
+                os.path.splitext(filename)[0] + ".appdata.xml",
+            )
+            with open(appdata, "w") as f:
+                f.write("""<application></application>""")
+
+            unsafe_mode = False
+            checker = ManifestChecker(manifest, unsafe_mode, require_important_update)
+            self.assertEqual(
+                len(checker.get_external_data()),
+                expected_data_count,
+            )
+
+            if test_important_update:
+                checker._checkers = [GitUpdateEverythingChecker]
+            else:
+                checker._checkers = [UpdateEverythingChecker]
+
+            await checker.check()
+            updates = checker.update_manifests()
+
+            with open(manifest, "r") as f:
+                new_contents = f.read()
+
+            self.assertEqual(new_contents, expected_new_contents)
+            if not test_important_update:
+                self.assertEqual(updates, expected_updates)
```

Comment:
Making this conditional seems like a bad idea. We should verify that exactly the updates we expected were applied.